src/synthetic_knn/synthetic_plots.py

```diff
@@ -0,0 +1,63 @@
+from __future__ import annotations
+
+import pandas as pd
+from numpy.typing import NDArray
+from sklearn.neighbors import NearestNeighbors
+
+from synthetic_knn.point_networks import PointNetwork
+
+
+class SyntheticPlots:
+    def __init__(
+        self, reference_coordinates: NDArray, network: PointNetwork, k: int = 10
+    ):
+        self.reference_coordinates = reference_coordinates
+        self.network = network.fit(reference_coordinates)
+        self.k = k
+        self.neighbors_ = self._generate_neighbors()
+
+    def synthetic_coordinates(self) -> NDArray:
+        return self.network.network_coordinates()
+
+    def _generate_neighbors(self) -> tuple[NDArray, NDArray]:
+        """Find the nearest reference neighbors of the synthetic mesh."""
+        nn_finder = NearestNeighbors(n_neighbors=self.k)
+        nn_finder.fit(self.reference_coordinates)
+        return nn_finder.kneighbors(self.synthetic_coordinates(), n_neighbors=self.k)
+
+    def distances(self) -> NDArray:
+        return self.neighbors_[0]
+
+    def neighbors(self, *, id_arr: NDArray = None) -> NDArray:
+        nn_idx = self.neighbors_[1]
+        return id_arr[nn_idx] if id_arr is not None else nn_idx
+
+    def write_synthetic_coordinates(
```

Comment:
Yes, I agree that this is a more flexible approach.  

I'm wondering if you have thoughts on the design here.  Is there an opportunity here to create a generic `to_frame` method for the `SyntheticPlots` class that could accommodate the translation of coordinates, neighbors, and distances, given that they will each specialize the dataframe slightly differently?   

Or is it just better to rename the current methods to make clear they return dataframes?  It feels weird to me to have a method, e.g., called `synthetic_coordinates` that returns an ndarray and another method called something like `synthetic_coordinates_as_frame` that returns a dataframe.

Reply:
>  It feels weird to me to have a method, e.g., called synthetic_coordinates that returns an ndarray and another method called something like synthetic_coordinates_as_frame that returns a dataframe.

Yeah, I agree. My first thought was to just add an `as_frame=False` parameter to `synthetic_coordinates`, `distances`, and `neighbors` so that each would return either an array or a formatted dataframe (sort of like the `sklearn.dataset` API), but I'm not convinced that's the best approach. 

> Is there an opportunity here to create a generic to_frame method for the SyntheticPlots class that could accommodate the translation of coordinates, neighbors, and distances, given that they will each specialize the dataframe slightly differently?

Great question. I like the idea of consolidating things and keeping return types consistent (unlike my idea above). Would you want `to_frame` to be able to produce each of the three dataframe types separately? I can't think of a great approach to tackle that - you could have parameters for whether to include the coordinates, distances, and/or neighbors, but that adds a little complexity if the users sets all the flags to false...

The simplest approach might be a `to_frame` method that just includes everything in one big dataframe, but that would complicate your workflow if you're usually writing them out to separate CSVs.

Reply:
Well, in the end, I've gone with option 1 and added `as_frame` options to these three methods (`synthetic_coordinates`, `distances`, and `neighbors`).  I created some private methods to handle duplication as well.

> The simplest approach might be a `to_frame` method that just includes everything in one big dataframe, but that would complicate your workflow if you're usually writing them out to separate CSVs.

I like this idea, but I felt the simplicity of writing out the information straight from the dataframes themselves was an important feature.  Please let me know if you disagree.

Reply:
> I like this idea, but I felt the simplicity of writing out the information straight from the dataframes themselves was an important feature. 

Yeah, makes perfect sense! I was a little worried about the duplication with the `as_frame` option, but I think your solution looks great.

Reply:
Excellent, thanks for your input here!